File: CleverFerretV2/feature/ai/src/main/java/com/cleverferret/v2/feature/ai/model/AiCapabilityFlags.java
package com.cleverferret.v2.feature.ai.model;

import java.util.EnumMap;
import java.util.Map;

public final class AiCapabilityFlags {
    private final Map<AiCapability, Boolean> flags;

    public AiCapabilityFlags(Map<AiCapability, Boolean> flags) {
        this.flags = new EnumMap<>(AiCapability.class);
        this.flags.putAll(flags);
    }

    public static AiCapabilityFlags allEnabled() {
        EnumMap<AiCapability, Boolean> values = new EnumMap<>(AiCapability.class);
        for (AiCapability capability : AiCapability.values()) {
            values.put(capability, true);
        }
        return new AiCapabilityFlags(values);
    }

    public boolean isEnabled(AiCapability capability) {
        return flags.getOrDefault(capability, false);
    }
}

File: CleverFerretV2/feature/ai/src/main/java/com/cleverferret/v2/feature/ai/model/AiCapabilityFlags.kt
package com.cleverferret.v2.feature.ai.model

class AiCapabilityFlags(flags: Map<AiCapability, Boolean>) {
    private val flags: Map<AiCapability, Boolean> = flags.toMap()

    fun isEnabled(capability: AiCapability): Boolean = flags.getOrDefault(capability, false)

    companion object {
        fun allEnabled(): AiCapabilityFlags = AiCapabilityFlags(AiCapability.entries.associateWith { true })
    }
}
